.gitignore

```diff
@@ -2,17 +2,146 @@
 #
 # SPDX-License-Identifier: MPL-2.0
 
-# Generated files
-*.egg-info
-
-# temporary_files
+# Byte-compiled / optimized / DLL files
 __pycache__/
-.mypy_cache/
-.pytest_cache/
+*.py[cod]
```

Comment:
@TonyXiang8787 This comment counts for all things that have changed in the `.gitignore`.

Why did you add so many things to ignore? I think this is an overkill. We should only add things to the gitignore when you plan to use is and don't want it to be commited. I think it doesn't make sense to try to add all the things that you can think of to the gitignore.

Reply:
The `.gitignore` is generated by GitHub itself. When you select a language, e.g. Python, GitHub will generate a standard `.gitignore` for common files which are ignored for Python project.

Reply:
Usually when you create a new repository from GitHub, it will generate a automatic `.gitignore` based on language selection.

Reply:
Not if you select .gitignore template: **None** 😛 , like I did 😉

Reply:
I'm not convinced of using the auto generated `.gitignore`. I think it looks messy to just have 100+ random things to ignore

Reply:
@bramstoeller @petersalemink95 it appears that both of you have similar opinion. I agree with you on this and you can delete most of the changes.